### PR TITLE
fixed named output nil case

### PR DESCRIPTION
**Pull request by applike-ss**

### CODE REVIEW

File: pkg/mdl/named.go [hunk 1]
```diff
@@ -13,6 +15,10 @@ type Nameable interface {
 }
 
 func NamedOutput(in interface{}) interface{} {
+	if in == nil || reflect.ValueOf(in).IsNil() {
```
Comment (ajscholl) [inline pkg/mdl/named.go:18]: IsNil will panic if `in` is not a chan, func, interface, map, pointer, or slice value

Comment (applike-ss) [reply pkg/mdl/named.go:18]: afaik, namedOutput should always get a pointer as of now

File: pkg/mdl/factory.go [hunk 2]
```diff
@@ -85,3 +88,29 @@ func EmptyUintIfNil(i *uint) uint {
...
+	switch v.Kind() {
+	case reflect.Chan,
+		reflect.Func,
+		reflect.Map,
+		reflect.Ptr,
+		reflect.UnsafePointer,
+		reflect.Interface,
+		reflect.Slice:
+	default:
+		return true
```
Comment (ajscholl) [inline pkg/mdl/factory.go:108]: Shouldn't this be `return false`? If you add some tests for this, I would expect that an empty struct is not nil, but I would also expect the current implementation would say it is nil.

File: pkg/mdl/factory.go [hunk 3]
```diff
@@ -85,3 +88,29 @@ func EmptyUintIfNil(i *uint) uint {
...
+	}
+
+	if v.IsNil() {
+		return true
```
Comment (ajscholl) [inline pkg/mdl/factory.go:112]: You can just `return v.IsNil()`

File: pkg/mdl/factory.go [hunk 4]
```diff
@@ -85,3 +88,29 @@ func EmptyUintIfNil(i *uint) uint {
...
+	v := reflect.ValueOf(m)
+
+	switch v.Kind() {
+	case reflect.Chan,
+		reflect.Func,
+		reflect.Map,
+		reflect.Ptr,
+		reflect.UnsafePointer,
+		reflect.Interface,
+		reflect.Slice:
```
Comment (ajscholl) [inline pkg/mdl/factory.go:106]: Actually, can we move `return v.IsNil()` here? Would be easier to read IMO

File: pkg/mdl/factory.go (hunk 2 above)
Comment (applike-ss) [reply pkg/mdl/factory.go:108]: i would rather want to check it properly, but what kind of types should i add checks for? don't want to make this a 100 liner
